### PR TITLE
Add dynamic collections

### PR DESCRIPTION
**Pull request by jlamontagne**
Hello!

Thanks for writing a great plugin. I've extended it with a feature I required.

Cheers,
Josh

---
### Dynamic collections
Enables use of `:symbols` that take on associated path segments and build
nested collections by grouping segments:

```js
collections({ departments: 'departments/:department/*.md' })
```

`src/departments/hardware/alice.md`
`src/departments/hardware/bob.md`
`src/departments/software/charlie.md`

The resulting collections can be nested or accessed directly:

```hbs
{{#each collections.departments}}
  {{department}}
  {{#each this}}
    {{{contents}}}
  {{/each}}
{{/each}}

{{#each collections.departments.hardware}}
  {{department}}
  {{{contents}}}
{{/each}}
```

### CODE REVIEW

**Comment by chadly** [comment]
:+1: this is exactly the feature I was looking for

File: test/fixtures/dynamic-frontmatter/src/one.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+collection: countries.canada.toronto
```
Comment (RobLoach) [inline test/fixtures/dynamic-frontmatter/src/one.md]: Yay Toronto!

**Comment by RobLoach** [comment]
This looks pretty great. Is there anything missing for a merge?

**Comment by madeofpeople** [comment]
I think this is definitely worth incorporating, 
@ianstormtaylor thoughts…

**Comment by ianstormtaylor** [comment]
This looks awesome. Sorry for the delay folks! I'll try to get a second this weekend to look this over and get a decision/merge.

**Comment by madeofpeople** [comment]
:+1:

**Comment by ianstormtaylor** [comment]
Hey @jlamontagne after looking through this, I think it's really cool, but it also should be built in a different plugin that is more opinionated. I'd like to keep this collections one much simpler.

**Comment by jme783** [comment]
@jlamontagne Is there any way to add sorting to dynamic collections? I would love to be able to control the order in which the dynamic collections render. Thanks!

**Comment by jlamontagne** [comment]
Thanks for the comments everyone! I'll create a separate plugin for this soon. I'll post a link here when it's done. @jme783 I'll add sorting options as well.

**Comment by RobLoach** [comment]
In the mean time, I've put together [`metalsmith-collection-convention`](https://github.com/robloach/metalsmith-collections-convention), which compliments collections by allowing you to define them directly from within files.

Thanks guys!

**Pull request closed without merging**